redux-form/lib/FieldArray.d.ts

```diff
@@ -20,24 +20,23 @@ interface BaseFieldArrayProps {
      *
      * Required but made optional so interface can be used on decorated components.
      */
-    component?: ComponentConstructor,
-    // component?: Component<P, any> | StatelessComponent<P>,
+    component?: ComponentConstructor<any>,
 
     /**
      * Allows you to to provide a field-level validation rule. The function will be given the
      * current value of the field and all the other form values. If the field is valid, it
      * should return `undefined`, if the field is invalid, it should return an error
      * (usually, but not necessarily, a `String`).
      */
-    validate?: Validator|Validator[];
+    validate?: Validator;
 
     /**
      * Allows you to to provide a field-level warning rule. The function will be given the
      * current value of the field and all the other form values. If the field needs a warning,
      * it should return the warning (usually, but not necessarily, a `String`). If the field
      * does not need a warning, it should return `undefined`.
      */
-    warn?: Validator|Validator[];
+    warn?: Validator;
```

Comment:
Both `validate` and `warn` can be a list of Validators
http://redux-form.com/6.5.0/examples/fieldLevelValidation/